docs/tutorials/algokit-template.md

```diff
@@ -0,0 +1,141 @@
+
+Example snippet:
+
+```json
+{
+  "name": "{{ project_name }}",
+  "version": "1.0.0",
+  "description": "{{ project_description }}",
+  "scripts": {
+    "build": "tsc"
+  },
+  "devDependencies": {
+    "typescript": "^4.2.4",
+    "tslint": "^6.1.3",
+    "tslint-config-prettier": "^1.18.0"
+  }
+}
+```
+
+This example shows how Jinja syntax is used within `package.json` to allow placeholders for project metadata and dependencies.
+
+### Bootstrap Option
+
+AlgoKit templates can include a bootstrap script. This script is automatically run after the project is initialized and can perform various setup tasks like installing dependencies or setting up databases.
+
+- `env`: Copies an `.env.template` file to `.env` in the current working directory and prompts for any unspecified values. This is a critical aspect of managing environment variables securely, ensuring that sensitive data doesn't accidentally end up in version control.
+- `poetry`: If your Python project uses Poetry for dependency management, the `poetry` command installs Poetry (if not present) and runs `poetry install` in the current working directory to install Python dependencies.
+- `npm`: If you're developing a JavaScript or TypeScript project, the `npm` command runs npm install in the current working directory to install Node.js dependencies.
+- `all`: The `all` command runs all the aforementioned bootstrap sub-commands in the current directory and its subdirectories. This command is a comprehensive way to ensure all project dependencies and environment variables are properly set up.
+
+### Predefined copier answers
+
+When initializing a new project, Copier can prompt the user for input, which is then passed to the template as variables. This is useful for customizing the new project based on user input.
+
+Example:
+
+```yaml
+# copier.yaml
+project_name:
+    type: str
+    help: What is the name of this project.
+    placeholder: "algorand-app"
+```
+This would prompt the user for the project name, and the input can then be used in the template using the Jinja syntax `{{ project_name }}`.
+
+#### Default Behaviors
+When creating an AlgoKit template, there are a few default behaviors that you can include to improve the development experience:
+
+- **Git**: If Git is installed on the user's system and the user's working directory is a Git repository, Copier will commit the newly created project as a new commit in the repository. This feature helps to maintain a clean version history for the project. If you wish to add a specific commit message for this action, you can specify a `commit_message` in the `_commit` option in your `copier.yaml` file.
+
+- **VSCode**: If the user has Visual Studio Code (VSCode) installed and the path to VSCode is added to their system's PATH, Copier can automatically open the newly created project in a new VSCode window. This is triggered by adding `vscode: true` in your `copier.yam` file.
```

Comment:
C̶o̶p̶i̶e̶r̶ -> AlgoKit CLI will automatically open the newly created VSCode window unless user provides specific flags into the init command